xesmf/smm.py

```diff
@@ -121,6 +122,15 @@ def apply_weights(weights, indata, shape_in, shape_out):
         Extra dimensions are the same as `indata`.
         If input data is C-ordered, output will also be C-ordered.
     """
+    # Limitation from numba : some big-endian dtypes are not supported.
+    try:
```

Comment:
how costly is this test? wouldn't it be cheaper to just test with:

```python
if indata.dtype.byteorder == '>'
```

Reply:
I don't think it's very costly. My intuition comes from the fact that it happens when numba doesn't support the given dtype, so no computation is done at all. But indeed,  it is usually less costly to perform a single check than a try-except call, that's from pure python.

However, I was trying to be as general as I can be here. This change was triggered by numba not supporting a given byte-order, but could there be other things that numba doesn't support, especially on other machines? Evidently, I can't test those with such a computer, but I tried to be ready for other eventualities...

Reply:
fair point, I'm ok with sacrificing a bit of performance for compatibility. try/except it is then